h2/stream.py

```diff
@@ -131,8 +132,13 @@ def request_sent(self, previous_state):
         Fires when a request is sent.
         """
         self.client = True
-        self.headers_sent = True
-        return []
+        if not self.headers_sent:
+            self.headers_sent = True
+            event = _HeadersSent()
+        else:
+            event = _TrailersSent()
```

Comment:
uh...actually, I think this branch is un-hittable. It looks a bit like maybe clients can't send trailers at all. That might be a separate bug.